libsolidity/analysis/NameAndTypeResolver.cpp

```diff
@@ -503,11 +503,20 @@ bool DeclarationRegistrationHelper::registerDeclaration(
 		else
 		{
 			auto shadowedLocation = shadowedDeclaration->location();
-			_errorReporter.warning(
-				2519_error,
-				_declaration.location(),
-				"This declaration shadows an existing declaration.",
-				SecondarySourceLocation().append("The shadowed declaration is here:", shadowedLocation)
+
+			if (shadowedDeclaration->visibility() == Visibility::External)
+				_errorReporter.warning(
+					8760_error,
+					_declaration.location(),
+					"This declaration has the same name as an existing external declaration.",
+					SecondarySourceLocation().append("The external declaration is here:", shadowedLocation)
+				);
```

Comment:
Is there a legitimate case when a declaration with `External` visibility is really shadowed by another declaration?